opal/mca/patcher/base/patcher_base_patch.c

```diff
@@ -106,6 +106,8 @@ static void flush_and_invalidate_cache (unsigned long a)
     __asm__ volatile("mfence;clflush %0;mfence" : :"m" (*(char*)a));
 #elif OPAL_ASSEMBLY_ARCH == OPAL_IA64
     __asm__ volatile ("fc %0;; sync.i;; srlz.i;;" : : "r"(a) : "memory");
+#elif OPAL_ASSEMBLY_ARCH == OPAL_ARM64
+    __asm__ volatile ("ic ialluis");
 #endif
```

Comment:
@hjelmn  If this is a self modifying code, I would expect to see ISB (or some stronger barrier) somewhere

Reply:
Ah, ok. Thats simple enough.

One of the checks failed for illegal instruction. I don't know where though. Is ```ic alluis``` ever considered privileged?

Reply:
SCTLR_EL1.UCI has to be set to 1 in order to enable cache maintenance operations in EL0 (user space). More generic question - what we are trying to do here ? Can you just use DSB ?

Reply:
DSB is probably all I need. Was just trying to replicate what was done on x86_64 which is most likely overkill.

Reply:
Will update to use DSB now.